src/query/storage/m3/dynamic_cluster_test.go

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) 2020  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/client"
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/retention"
+	"github.com/m3db/m3/src/query/storage/m3/storagemetadata"
+	"github.com/m3db/m3/src/x/ident"
+	"github.com/m3db/m3/src/x/instrument"
+	xtest "github.com/m3db/m3/src/x/test"
+	xwatch "github.com/m3db/m3/src/x/watch"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	defaultTestNs1ID         = ident.StringID("testns1")
+	defaultTestNs2ID         = ident.StringID("testns2")
+	defaultTestRetentionOpts = retention.NewOptions().
+					SetBufferFuture(10 * time.Minute).
+					SetBufferPast(10 * time.Minute).
+					SetBlockSize(2 * time.Hour).
+					SetRetentionPeriod(48 * time.Hour)
+	defaultTestNs2RetentionOpts = retention.NewOptions().
+					SetBufferFuture(10 * time.Minute).
+					SetBufferPast(10 * time.Minute).
+					SetBlockSize(4 * time.Hour).
+					SetRetentionPeriod(48 * time.Hour)
```

Comment:
Are these meant to match `defaultTestRetentionOpts` except blockSize? this can just be `defaultTestRetentionOpts.SetBlockSize(4 * time.Hour)` if so